ext/mysqli/mysqli_warning.c

```diff
@@ -254,67 +254,12 @@ static int mysqli_warning_errno(mysqli_object *obj, zval *retval, zend_bool quie
 }
 /* }}} */
 
-/* {{{ mysqli_warning_construct(object obj) */
 PHP_METHOD(mysqli_warning, __construct)
 {
-	zval			*z;
-	mysqli_object	*obj;
-#ifndef MYSQLI_USE_MYSQLND
-	MYSQL			*hdl;
-#endif
-	MYSQLI_WARNING  *w;
-	MYSQLI_RESOURCE *mysqli_resource;
-
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "o", &z) == FAILURE) {
-		RETURN_THROWS();
-	}
-	obj = Z_MYSQLI_P(z);
-
-	if (obj->zo.ce == mysqli_link_class_entry) {
-		MY_MYSQL *mysql;
-		MYSQLI_FETCH_RESOURCE_CONN(mysql, z, MYSQLI_STATUS_VALID);
-		if (mysql_warning_count(mysql->mysql)) {
-#ifndef MYSQLI_USE_MYSQLND
-			w = php_get_warnings(mysql->mysql);
-#else
-			w = php_get_warnings(mysql->mysql->data);
-#endif
-		} else {
-			php_error_docref(NULL, E_WARNING, "No warnings found");
-			RETURN_FALSE;
-		}
-	} else if (obj->zo.ce == mysqli_stmt_class_entry) {
-		MY_STMT *stmt;
-		MYSQLI_FETCH_RESOURCE_STMT(stmt, z, MYSQLI_STATUS_VALID);
-#ifndef MYSQLI_USE_MYSQLND
-		hdl = mysqli_stmt_get_connection(stmt->stmt);
-		if (mysql_warning_count(hdl)) {
-			w = php_get_warnings(hdl);
-#else
-		if (mysqlnd_stmt_warning_count(stmt->stmt)) {
-			w = php_get_warnings(mysqli_stmt_get_connection(stmt->stmt));
-#endif
-		} else {
-			php_error_docref(NULL, E_WARNING, "No warnings found");
-			RETURN_FALSE;
-		}
-	} else {
-		php_error_docref(NULL, E_WARNING, "Invalid class argument");
-		RETURN_FALSE;
```

Comment:
The warning + return false is how I got here ... a constructor is not allowed to return false.